src/easycrypt/syntax/statement/mod.rs

```diff
@@ -0,0 +1,118 @@
+//!
+//! EasyCrypt AST nodes containing statements. Statements are a syntactic
+//! category whose computations result in actions, potentially with
+//! side-effects.
+//!
+
+pub mod block;
+pub mod call;
+pub mod if_conditional;
+pub mod while_loop;
+
+use self::block::Block;
+use self::call::ProcCall;
+use self::if_conditional::IfConditional;
+use self::while_loop::WhileLoop;
+
+use crate::easycrypt::syntax::definition::Definition;
+use crate::easycrypt::syntax::expression::Expression;
+use crate::easycrypt::syntax::reference::Reference;
+
+/// EasyCrypt AST nodes containing statements. Statements are a syntactic
```

Comment:
Well, if EasyCrypt is not a strict subset of Yul, it makes sense to redefine the whole AST.
Is it correct? Could you add the link to the AST spec to PR description?

Reply:
It is not a subset; for once, there are functions and procedures, there are binary operations, `else`, no `for` loop but `while` loop etc.
I have added a link to the EasyCrypt reference to the PR and once we move the code into a separate repository I ll add a short description of the supported AST there directly.